File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/storage/disk/RecordsStorage.java
package com.worldventures.dreamtrips.wallet.domain.storage.disk;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.worldventures.dreamtrips.wallet.domain.entity.record.Record;

import java.util.List;

public interface RecordsStorage extends ModelStorage {

   void saveRecords(List<Record> items);

   @NonNull
   List<Record> readRecords();

   void deleteAllRecords();

   void saveDefaultRecordId(String id);

   @Nullable
   String readDefaultRecordId();

   void deleteDefaultRecordId();

   void saveOfflineModeState(boolean enabled);

   @NonNull
   Boolean readOfflineModeState();

}

File: wallet/src/main/java/com/worldventures/wallet/domain/storage/disk/RecordsStorage.kt
package com.worldventures.wallet.domain.storage.disk

import com.worldventures.wallet.domain.entity.record.Record

interface RecordsStorage : ModelStorage {

   fun saveRecords(items: List<Record>?)

   fun readRecords(): List<Record>

   fun deleteAllRecords()

   // default card
   fun saveDefaultRecordId(id: String?)

   fun readDefaultRecordId(): String?

   fun deleteDefaultRecordId()

   // offline mode
   fun saveOfflineModeState(enabled: Boolean)

   fun readOfflineModeState(): Boolean
}
